City-of-Helsinki/hel-fi-drupal-grants
# feat: AU-2015: Debug mode for E2E tests

# [AU-2015](https://helsinkisolutionoffice.atlassian.net/browse/AU-2015)

## What was done
This pull request implements a debugging mode for the E2E tests. The debugging mode can be enabled by adding the following line at the end of the `.test_env` file: `APP_DEBUG=TRUE` (note that no quotation marks are used).

Furthermore, the debugging mode has been used to implement a wrapper for the `console.log` command. When logging messages with with the new `logger()` function, messages will only be logged to the console if the debugging mode has been enabled.

In addition to this, all calls for `console.log` have been replaced with calls to the new `logger()` function.

## Thoughts
The `isDebuggingEnabled()` function inside the `debugging_helpers.ts` file could potentially be used when implementing other features that should only be available when debugging has been enabled. One example could be the `slowLocator()` function inside the `helpers.ts` file.

## How to install
Make sure your instance is up and running on the correct branch.
  * `git checkout feature/AU-2015-e2e-debug-mode`
  * `make fresh`
  * `make drush-cr`

## How to test
- Run some tests without having enabled the debugging mode, for example: `make test-pw-p PROJECT=forms-29`. The output should look something like this:

![Screenshot 2024-01-11 at 10 36 39](https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/89e4f0f4-3c35-4be8-b3e0-5ecbd64a2519)

- Now, add `APP_DEBUG=TRUE` to the end of the `.test_env` file and run `make test-pw-p PROJECT=forms-29` again. The output should look something like this:

![Screenshot 2024-01-11 at 10 33 40](https://github.com/City-of-Helsinki/hel-fi-drupal-grants/assets/117807015/74774b6b-3d8b-4f37-89fb-98338c29bebb)

- Review the code, especially `debugging_helpers.ts` and `logger.ts`.


[AU-2015]: https://helsinkisolutionoffice.atlassian.net/browse/AU-2015?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
